examples/single_page_app/custom_sub_pages.py

```diff
@@ -0,0 +1,70 @@
+from typing import Callable
+
+from nicegui import app, ui
+from nicegui.page_args import RouteMatch
+
+
+def protected(func: Callable) -> Callable:
+    """Decorator to mark a route handler as requiring authentication for the custom_sub_pages."""
+    func._is_protected = True  # pylint: disable=protected-access
+    return func
+
+
+class CustomSubPages(ui.sub_pages):
+    """Custom ui.sub_pages with built-in authentication and custom 404 handling."""
+
+    def _show_page(self, match: RouteMatch) -> None:
+        if self._is_route_protected(match.builder):
+            if not self._is_authenticated():
+                self._show_login_form(match.full_url)
+                return
+        super()._show_page(match)
```

Comment:
I think this method is called `_render_page` now.

Reply:
I'm sorry. It's fixed now in 39986e7.